cmd/core/check-payload-value.go

```diff
@@ -48,12 +48,18 @@ var checkPayloadValueCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		var err error
 
-		log.Infof("Using eth node: %s", ethNodeURI)
 		client := flashbotsrpc.New(ethNodeURI)
-		var client2 *flashbotsrpc.FlashbotsRPC
+		if client == nil {
```

Comment:
can this ever be nil?

Reply:
with the current implementation of `flashbotsrpc` library it's never `nil`.  but here, as a user, I can not make that assumption because if it's never `nil``, why not just return the struct itself?

Reply:
> why not just return the struct itself

you mean the value instead of a pointer?

Reply:
yes.

I mean: if I treat the lib as a black-box then I cannot know that the returned pointer is never nil, moreover the fact that the author of the lib decide to return pointer and not just the value suggests that perhaps there are cases when it can be nil.

also, there is a question of future-proof, etc, etc.